relay-server/src/services/spooler/mod.rs

```diff
@@ -1029,13 +1098,49 @@ impl BufferService {
         Ok(())
     }
 
+    /// Handles retrieving of the index from the underlying spool.
+    ///
+    /// To process this request we spawn the tokio task and once it's done the result will be sent
+    /// to [`ProjectCache`].
+    ///
+    /// If the spool is memory based, we ignore this request - the index in the
+    /// [`ProjectCache`] should be full and correct.
+    /// If the spool is located on the disk, we read up the keys and compile the index of the
+    /// spooled envelopes for [`ProjectCache`].
+    async fn handle_get_index(&mut self, _: RestoreIndex) -> Result<(), BufferError> {
```

Comment:
Should we rename this to `handle_restore_index` as well?

Reply:
yep, will do it now